### DISCUSSION TITLE
Download/save query results

### DISCUSSION THREAD

**Discussion by ronyarmon** [duckdb/duckdb-wasm#1001] [Ideas]
I'm a wasm newbie that reached your great tool while looking for an SQL engine that is able to query parquet files. The analysts querying this files are not using Python of CLI so DuckDB shell seems a great option to query the files I produced. However, I did not see any option to download or save the query results. Is there such an option that I'm missing?

**Comment by Mause** [comment]
If you're looking to export as csv or parquet you can do that by writing the file within a query, then reading the file from the filesystem using JavaScript

**Reply by ronyarmon** [reply]
Thanks, I was looking for a no-code solution that will be suitable for analysts who are used to conventional database viewers. I guess it's not available but it may be a useful future feature to add to DuckDB

**Reply by ankoh** [reply]
we have this, if you want to stay in the web shell:
![shell duckdb org_ (4)](https://user-images.githubusercontent.com/3986510/189501075-f8d3940c-69fd-4796-9522-2b5790020d40.png)

**Reply by ankoh** [reply]
.files download $filename will open the file save dialog of the browser

**Reply by ronyarmon** [reply]
So fast, simple and clear! Many thanks for this useful tool and response
